app/controllers/mobile.js

```diff
@@ -0,0 +1,73 @@
+import Controller from '@ember/controller';
+import { action } from '@ember/object';
+import { inject as service } from '@ember/service';
+import { ERROR_MESSAGES } from '../constants/error-messages';
+import {
+  AUTH_STATUS,
+  MOBILE_LOGIN_SUCCESS_MESSAGE,
+  QR_SCAN_CONFIRMATION_MESSAGE,
+  QR_SCAN_MESSAGE,
+  REQUEST_CANCEL_MESSAGE,
+} from '../constants/auth-status';
+import {
+  QR_AUTHORIZATION_STATUS_URL,
+  USER_AUTHENTICATED_DEVICES_URL,
+} from '../constants/apis';
+import { TOAST_OPTIONS } from '../constants/toast-options';
+
+export default class MobileController extends Controller {
+  @service toast;
+  @service router;
+
+  async updateQRAuthStatus(status, successMessage) {
+    try {
+      const response = await fetch(`${QR_AUTHORIZATION_STATUS_URL}/${status}`, {
+        method: 'PATCH',
+        headers: {
+          'Content-Type': 'application/json',
+        },
+        credentials: 'include',
+      });
```

Comment:
NIT: Please create a util function for fetching, where we just pass the method, body and headers if required